### ISSUE TITLE
how to using mouse_event with Aimbot and RCS.

### ISSUE THREAD

**Issue by Eval0** [TKazer/CS2_External#31]
how to using mouse_event with Aimbot and RCS.

**Comment by goliat26** [comment]
Is memory writing bannable? If not, why you cant use write viewAngle.
I dont know if is it, i dont use this software but i only question.

**Comment by TKazer** [comment]
> Is memory writing bannable? If not, why you cant use write viewAngle. I dont know if is it, i dont use this software but i only question.

no, it will no cause ban. maybe he just wanted to be more safer.

**Comment by Eval0** [comment]
> Is memory writing bannable? If not, why you cant use write viewAngle. I dont know if is it, i dont use this software but i only question.

if keep cheat privare should be fine,but use WPM  is easy VAC dedtectable,so i use my driver only read memory but idk how to using mouse_event with aimbot and RCS

**Comment by goliat26** [comment]
> if keep cheat privare should be fine,but use WPM is easy VAC dedtectable,so i use my driver only read memory but idk how to using mouse_event with aimbot and RCS

Only read + driver read sounds mega safety :D. Do you use just MmCopyVirtualMemory in driver? And which um<->km communication type?

About mouse_event... Only MOUSEEVENTF_MOVE flag works. MOUSEEVENTF_ABSOLUTE flag doesn't. mouse_event depends on game sensitivity, and Windows mouse sensitivity so you need something like "factor" to adjust by player. Last think is that mouse_event is only moving mouse by x and y pixels it is not accuracy like view angle for long distance to enemy.

You have to calculate "WorldToScreen" of enemy's head (or another bone) position and calculate difference from actual mouse position (most time center of screen) to it and move mouse to this position.

**Comment by Sigma12456** [comment]
I also am going for a safe way. I implemented this cheat into my own driver, not writing memory using a bit different function then mouse event and yes you NEED MmCopyVirtualMemory. even tho its undocumented you can find everything you need on the internet. Also RCS is possible you can convert the yaw and pitch to screen positions and move the mouse step by step. just reversing the same rcs function as the memory write thing

**Comment by goliat26** [comment]
I have implemented mouse event based aimbot but it is not accuracy like view angle setting. Especially with full auto shooting. With tapping it works fine and with correct "factor" value (because mouse moving depends on mouse sens) it is fast. Meaby have you know how to read mouse sens from memory?

**Comment by Cliqnt** [comment]
> I have implemented mouse event based aimbot but it is not accuracy like view angle setting. Especially with full auto shooting. With tapping it works fine and with correct "factor" value (because mouse moving depends on mouse sens) it is fast. Meaby have you know how to read mouse sens from memory?

hi can we talk on discord please im trynna do the same thing as him with arduino / raspberry to stop using WPM 
Reese#0613

**Comment by goliat26** [comment]
> > I have implemented mouse event based aimbot but it is not accuracy like view angle setting. Especially with full auto shooting. With tapping it works fine and with correct "factor" value (because mouse moving depends on mouse sens) it is fast. Meaby have you know how to read mouse sens from memory?
> 
> hi can we talk on discord please im trynna do the same thing as him with arduino / raspberry to stop using WPM Reese#0613

You don't have to use external device like arduino. You can just use mouse_event function to move mouse/crosshair. Why you need to use arduino/raspberry?

**Comment by Cliqnt** [comment]
> > > I have implemented mouse event based aimbot but it is not accuracy like view angle setting. Especially with full auto shooting. With tapping it works fine and with correct "factor" value (because mouse moving depends on mouse sens) it is fast. Meaby have you know how to read mouse sens from memory?
> > 
> > 
> > hi can we talk on discord please im trynna do the same thing as him with arduino / raspberry to stop using WPM Reese#0613
> 
> You don't have to use external device like arduino. You can just use mouse_event function to move mouse/crosshair. Why you need to use arduino/raspberry?

tought that it would be flagged / patched soon. can you add me on discord i really wanna know more about this please
